src/python/torchdistx/deferred_init.py

```diff
@@ -42,6 +48,11 @@ def materialize_tensor(tensor: Tensor) -> Tensor:
     Args:
         tensor:
             The tensor instance to materialize.
+
+    .. Warning::
+        Once materialized a fake tensor will hold a reference to its
+        materialized version. In order to avoid memory leaks make sure to
+        dispose it when it is no longer required.
```

Comment:
Is there any special handling required to avoid memory leaks compared to non-fake tensors? If so, would be good to document those possibly with an example.

Reply:
For `materialize_module()` there is no special handling required since it already overwrites the parameters of the model which practically disposes the fake tensors. For `materialize_tensor()` one has to be careful to not keep the fake tensor around if its materialized version is not needed anymore since the fake tensor internally holds a reference to it.